### PR TITLE
Add preact-testing-library and UI tests for TryExtensionLinkDialog

### PR DESCRIPTION
**Pull request by vishnumad**
### *Summary*

* Adds preact testing library
* Adds ui tests for TryExtensionLinkDialog

### CODE REVIEW

File: src/components/LinkDialog-css.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
...
...dialog-container-dark .-cbwsdk-link-dialog-box-content-qrcode-connecting p{color:#ddd}.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-content a,.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-content a:link,.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-content a:visited{color:#888}.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-content a:hover,.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-content a:active{color:#aaa}.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-cancel{background-color:#333}.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-cancel-x::before,.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-cancel-x::after{background-color:#aaa}.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-cancel:hover .-cbwsdk-link-dialog-box-cancel-x::before,.-cbwsdk-css-reset .-cbwsdk-link-dialog-container-dark .-cbwsdk-link-dialog-box-cancel:hover .-cbwsdk-link-dialog-box-cancel-x::after{background-color:#eee}`
```
Comment (ljharb) [inline src/components/LinkDialog-css.ts:4]: these look generated, why are they checked in?

File: src/components/TryExtensionLinkDialog.tsx [hunk 2]
```diff
@@ -122,7 +122,11 @@ const TryExtensionBox: FunctionComponent<{
             again.
           </div>
         )}
-        <button type="button" onClick={clickHandler}>
+        <button
+          data-testid="try-extension-button"
```
Comment (ljharb) [inline src/components/TryExtensionLinkDialog.tsx:126]: oof, testIDs are a brittle way to do testing; is there not a more robust mechanism to find the only button here?

File: src/components/LinkDialog-css.ts (hunk 1 above)
Comment (vishnumad) [reply src/components/LinkDialog-css.ts:4]: Do you recommend running `compile-assets.js` as part of the testing step instead?

Comment (ljharb) [reply src/components/LinkDialog-css.ts]: Absolutely yes. Generated files should never be committed to version control.

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
   "author": "Coinbase, Inc.",
   "license": "Apache-2.0",
   "scripts": {
-    "test:unit": "jest",
+    "test:unit": "node compile-assets.js && jest",
```
Comment (ljharb) [inline package.json:26]:
```suggestion
    "pretest:unit": "node compile-assets.js",
    "test:unit": "jest",
```

File: src/components/TryExtensionLinkDialog.test.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/preact";
+import { h } from "preact";
+
+import { TryExtensionLinkDialog } from "./TryExtensionLinkDialog";
+
+const renderTryExtensionLinkDialog = ({
+  connectDisabled = false,
+  isConnected = true
```
Comment (ljharb) [inline src/components/TryExtensionLinkDialog.test.tsx:8]:
```suggestion
  isConnected = true,
```
i'd expect the linter to fail on this tbh

Comment (vishnumad) [reply src/components/TryExtensionLinkDialog.test.tsx:8]: The trailing commas option is set to none in prettier configuration so it just gets auto removed

File: package.json (hunk 3 above)
Comment (vishnumad) [reply package.json]: Nice, didn't know this was a thing

File: src/components/TryExtensionLinkDialog.test.tsx (hunk 4 above)
Comment (erin-at-work) [reply src/components/TryExtensionLinkDialog.test.tsx:8]: We can set trailing commas to `true` on a separate PR, I also prefer it on

**Review by erin-at-work — APPROVED** [review]
lgtm!

**Comment by cb-heimdall** [comment]
Review Error for cb-jake @ 2022-05-02 17:31:27 UTC
User failed mfa authentication, public email is not set on your github profile. see [go/mfa-help](http://go/mfa-help)

**Review by cb-jake — APPROVED** [review]

File: src/components/TryExtensionLinkDialog.test.tsx (hunk 4 above)
Comment (Lojzicek) [reply src/components/TryExtensionLinkDialog.test.tsx:8]: true

**Pull request merged**